### PR TITLE
Run wire-format verification (tests client against reference server)

### PR DESCRIPTION
**Pull request by iamdanfox**
## Before this PR

conjure-java-jaxrs-clients were intended to adhere to the conjure wire spec, but tests had to be manually written to enforce this.

## After this PR

We exercise a conjure-java-jaxrs-client against the reference server from https://github.com/palantir/conjure-verification, using the standardized set of test cases.

Follow-up work:

- also exercise the conjure-java-jersey-server against a reference client
- test the retrofit client

### CODE REVIEW

File: conjure-java-verifier/src/test/java/com/palantir/verification/ExampleTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+package com.palantir.verification;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.conjure.verification.AutoDeserializeService;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class ExampleTest {
```
Comment (ferozco) [inline conjure-java-verifier/src/test/java/com/palantir/verification/ExampleTest.java]: do we still need this checked in? I assume this was for getting the tests setup

File: conjure-java-verifier/build.gradle [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+task conjureJavaJersey(type: Exec, dependsOn: unpackGenerator) {
+    executable "${-> unpackGenerator.getExecutable()}"
+    args 'generate',  "${-> configurations.verificationApi.singleFile}", 'src/generatedJersey/java', '--jersey'
+
+    inputs.file "${-> configurations.verificationApi.singleFile}"
+    outputs.dir "src/generatedJersey/java"
+    doFirst { delete "src/generatedJersey/java/*" }
+}
+
+task conjureJavaRetrofit(type: Exec, dependsOn: unpackGenerator) {
```
Comment (ferozco) [inline conjure-java-verifier/build.gradle]: why do we generate retrofit?

**Comment by ferozco** [comment]
oh also, can we make the project `conjure-java-client-verifier`?

File: conjure-java-verifier/build.gradle (hunk 2 above)
Comment (dansanduleac) [reply conjure-java-verifier/build.gradle]: we want to test it!

Comment (dansanduleac) [reply conjure-java-verifier/build.gradle]: actually we should test it in conjure-java as well

**Review by ferozco — APPROVED** [review]

**Pull request merged**